python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,228 @@
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map 
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+        
+    def _get_rc(self, env=None):
+        rc = {}
+        
+        if env:
+            rcfile_location = env["GISRC"]
+        else:
+            rcfile_location = os.environ["GISRC"]
+            
+        with open(rcfile_location) as f:
+            for line in f:
+                entry = line.strip("\n").split(": ")
+                rc[entry[0]] = entry[1]
+        return rc
+    
+    
+
+    def _convert_coordinates(self, coordinates, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+        extent_ne = "{}, {}".format(extent["e"], extent["n"])
+        extent_sw = "{}, {}".format(extent["w"], extent["s"])
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent_ne, proj)
+        south, west = self._convert_coordinates(extent_sw, proj)
+
+        extent = {'North': north, 'South': south, 'East': east, 'West': west}
+
+        return extent
+    
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        
+        bounding_box = [[extent['North'], extent['West']],
+                        [extent['South'], extent['East']]]
+        
+        return bounding_box
+    
+        
+    def add_vector(self, vector, mapset=None):
+        """Imports vector into temporary WGS84 location, 
+        re-formats to a GeoJSON and adds to folium map"""
+        if mapset is None:
+            mapset = self._rc_original["MAPSET"]
```

Comment:
This will actually break for more advanced cases. Adding mapset is actually more sophisticated. If v.proj can handle that, that would be best.

Reply:
Hmm, if I leave the mapset kwarg empty in v.proj, I get an error that the vector cannot be found in "user1" (because the vectors I've been using are all in the PERMANENT mapset). Is this what you had in mind? Not sure how else to do this...

Reply:
Right, v.proj can't do this automatically in this context. It just defaults to the current mapset (and it is suspicious that it is user1 in your case, but that's a separate issue).

You need to use `gs.find_file` which will do the correct lookup for you. This needs to run in the current mapset and than the mapset name you acquire will go to v.proj running in that WGS84 location.

Reply:
Ah! Ok, that makes sense. I think I have it working with `gs.find_file` now.

It is "user1" because that's what I called in gj.init.